larray/core/array.py

```diff
@@ -6112,8 +6121,11 @@ def set_labels(self, axis, labels=None, inplace=False):
         """
         if isinstance(axis, dict):
             changes = axis
-        else:
+        elif isinstance(axis, (basestring, Axis, int)):
             changes = {axis: labels}
+        else:
```

Comment:
We probably want to add a something like: `assert axis is None` here. Otherwise errors could pass silently. eg
a user might write: arr.set_labels(['a', 'b'], ['a0,a1', 'b0,b1']) and I think this would silently pass (but do nothing).